src/Storages/MergeTree/MergeTreeData.cpp

```diff
@@ -2126,6 +2139,11 @@ void MergeTreeData::loadDataParts(bool skip_sanity_checks, std::optional<std::un
             }
 
             bool is_disk_defined = defined_disk_names.contains(disk_name);
+            if (!is_disk_defined && !lookOnDisk(disk))
```

Comment:
Is it possible the disk is "defined" but we still don't want to look at it? sorry for the dumb question, what does "is_disk_defined" actually mean?

1. Disk is defined somewhere in the settings regardless of being related to this table;
2. Disk is defined AND is related to this table